lxd/images.go

```diff
@@ -310,6 +310,13 @@ func imgPostInstanceInfo(d *Daemon, r *http.Request, req api.ImagesPost, op *ope
 	writer = shared.NewQuotaWriter(writer, budget)
 	meta, err = c.Export(writer, req.Properties)
```

Comment:
This won't work. As I mentioned in the GH issue, you need to pass the expiry through to `Export()` so that the `metadata.yaml` file inside the image can get the expiry too.

Reply:
https://github.com/lxc/lxd/issues/8808#issuecomment-850707063

Reply:
With what you have done now, if I do:
 - lxc publish blah --expire SOME-DATE --alias foo
 - lxc image export foo
 - lxc image delete foo
 - lxc image import foo.tar.xz

The image will no longer have an expiry as the only place which was overriden was the database, the expiry never making it into the image's metadata itself.

Reply:
Ah, I misunderstood, I thought we actually wanted the expiration date to be tied only to the database, not the image itself. Also that we only wanted `publish` to be able to read the metadata, not write to it! 
I'll fix that :)

Reply:
If `--expire` is specified, it should override anything in the current metadata.
If it's not specified, we'll use the existing expiry in the metadata or if none is set, fallback to never expiring.

Reply:
override _and_ overwrite!